tests/__init__.py

```diff
@@ -295,7 +296,7 @@ def server_socket_thread(srv):
                 )
         except Exception as e:
             # traceback.print_exc caused IOError: concurrent operation on sys.stderr.close() under setup.py test
-            sys.stderr.write(traceback.format_exc().encode())
+            print(traceback.format_exc(), file=sys.stderr)
```

Comment:
logging.warning or .error?

Reply:
It's test infrastructure. I'd rather avoid logging there. Do you insist?

Reply:
Print statements are sort of the worst for everything. It's also for consistency.

What's the intention for printing the error? Perhaps just delete this line?

Reply:
Yes, intention is not clear. It seems I suck at context managers with regard to error handling. Problem summarized in this pseudo-snippet:

```
@contextmanager
def helper():
  shared = [None]  # cross-thread variable sharing
  t = Thread(catches_exception_into_shared)
  t.start()
  yield "to test code"
  # anything below is not executed
  t.join()
  if shared[0]:
    raise shared[0]  # so raise is also not executed
```